apps/src/lib/client/rpc.rs

```diff
@@ -0,0 +1,152 @@
+//! Client RPC queries
+
+use std::io::{self, Write};
+
+use anoma::types::{address, storage, token};
+use borsh::BorshDeserialize;
+use tendermint_rpc::{Client, HttpClient};
+
+use crate::cli::args;
+use crate::node::ledger::rpc::{Path, PrefixValue};
+
+/// Dry run a transaction
+pub async fn dry_run_tx(
+    ledger_address: &tendermint::net::Address,
+    tx_bytes: Vec<u8>,
+) {
+    let client = HttpClient::new(ledger_address.clone()).unwrap();
+    let path = Path::DryRunTx;
+    let response = client
+        .abci_query(Some(path.into()), tx_bytes, None, false)
+        .await
+        .unwrap();
+    println!("{:#?}", response);
+}
+
+/// Query token balance(s)
+pub async fn query_balance(args: args::QueryBalance) {
+    let client = HttpClient::new(args.query.ledger_address).unwrap();
+    let tokens = address::tokens();
+    match (args.token.as_ref(), args.owner.as_ref()) {
+        (Some(token), Some(owner)) => {
+            let key = token::balance_key(token, owner);
+            let balance: token::Amount = query_storage_value(client, key).await;
+            let currency_code = tokens
+                .get(token)
+                .cloned()
+                .unwrap_or_else(|| token.to_string());
+            println!("{}: {}", currency_code, balance);
+        }
+        (None, Some(owner)) => {
+            for (token, currency_code) in tokens {
+                let key = token::balance_key(&token, owner);
+                let balance: token::Amount =
+                    query_storage_value(client.clone(), key).await;
+                println!("{}: {}", currency_code, balance);
+            }
+        }
+        (Some(token), None) => {
+            let key = token::balance_prefix(token);
+            let balances =
+                query_storage_prefix::<token::Amount>(client, key).await;
+            let currency_code = tokens
+                .get(token)
+                .cloned()
+                .unwrap_or_else(|| token.to_string());
+            let stdout = io::stdout();
+            let mut w = stdout.lock();
+            writeln!(w, "Token {}:", currency_code).unwrap();
```

Comment:
why the switch from `println!` to `writeln!` on stdout

Reply:
it's because it's printing in a loop to avoid system stdout lock/unlock on every statement